haystack_experimental/components/writers/chat_message_writer.py

```diff
@@ -0,0 +1,109 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import importlib
+from typing import Any, Dict, List
+
+from haystack import DeserializationError, component, default_from_dict, default_to_dict, logging
+from haystack.dataclasses import ChatMessage
+
+from haystack_experimental.chat_message_stores.types import ChatMessageStore
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class ChatMessageWriter:
+    """
+    Writes chat messages to an underlying ChatMessageStore.
+
+    Usage example:
+    ```python
+    from haystack.dataclasses import ChatMessage
+    from haystack_experimental.components.writers import ChatMessageWriter
+    from haystack_experimental.chat_message_stores.in_memory import InMemoryChatMessageStore
+
+    messages = [
+        ChatMessage.from_assistant("Hello, how can I help you?"),
+        ChatMessage.from_user("I have a question about Python."),
+    ]
+    message_store = InMemoryChatMessageStore()
+    writer = ChatMessageWriter(message_store)
+    writer.run(messages)
+    ```
+    """
+
+    def __init__(self, message_store: ChatMessageStore):
+        """
+        Create a ChatMessageWriter component.
+
+        :param message_store:
+            The ChatMessageStore where the chat messages are to be written.
+        """
+        self.message_store = message_store
+
+    def _get_telemetry_data(self) -> Dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        return {"message_store": type(self.message_store).__name__}
```

Comment:
I don't see this as being useful right now (since there's only one message store)? Let's not add telemetry data unless and until it makes sense.